src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.cs

```diff
@@ -1100,18 +1098,6 @@ private void BindHelper(NetworkCredential newCredential, bool needSetCredential)
             {
                 error = LdapPal.BindToDirectory(_ldapHandle, null, null);
             }
-            else if (AuthType == AuthType.Basic)
```

Comment:
In here we were already doing the append of the domain if authtype was basic, and this was happening in Linux as well so not sure why you moved that into Internal Bind of each platform instead.

Reply:
I've moved to Internal, because in Linux always seems to use `ldap_simple_bind`. 

```
BindToDirectory(ConnectionHandle ld, string who, string passwd) => Interop.ldap_simple_bind(ld, who, passwd);
```

Inside Linux if not External or Kerberos will run the same method above:
```
if (tempCredential == null && (AuthType == AuthType.External || AuthType == AuthType.Kerberos))
//
else
{
    error = Interop.ldap_simple_bind(_ldapHandle, cred.user, cred.password);
}
```

So, you don't think it is reasonable to move it into here? If you run out the same method, why not write it once and the same rules? And after all libldap only use AUTH_SIMPLE under the hood, which mean that doesn't matter what AuthType the user have choosed.

These was my decisions behind this code.